src/workloads/query/BooleanSimplifier.yml

```diff
@@ -0,0 +1,208 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query-optimization"
+
+Description: |
+  This workload measures performance of boolean expressions which can be simplified by
+  the Boolean Simplifier. 
+
+RandomSeed: 172547222
+
+GlobalDefaults:
+  Database: &Database test
+  DocumentCount: &DocumentCount 1e5
+  Threads: &Threads 1
+  MaxPhases: &MaxPhases 8
+  Duration: &Duration 2 seconds
+  QueryStringIn: &QueryStringIn [ "HrHCEtBAGnv",
+                                  "pCqgFttJBou",
+                                  "tjOnAuHAOpi",
+                                  "PmqoJGpDFjB",
+                                  "jEkEGEBnICu",
+                                  "kjPIJuEMGjm",
+                                  "nBKiBqKPCJN",
+                                  "BFkCIrIECBt",
+                                  "glhsCMqJPEs",
+                                  "lEMoKmmCiKv"
+                                ]
+
+Actors:
+- Name: ClearCollection
+  Type: CrudActor
+  Database: *Database
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [0]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+        Collection: Collection0
+        Operations:
+        - OperationName: drop
+
+- Name: InsertData
+  Type: Loader
+  Threads: 4
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *Database
+        MultipleThreadsPerCollection: true
+        CollectionCount: 1
+        DocumentCount: *DocumentCount
+        BatchSize: 1000
+        Document:
+          a1: {^RandomInt: {min: 0, max: 100}}
+          b1: {^Inc: {start: 11, step: 2}}
+          a2: {^RandomInt: {min: 0, max: 1000}}
+          b2: 3
+          c2: {^RandomInt: {min: 0, max: 5}}
+          d2: {^RandomInt: {min: 0, max: 5}}
+          e2: 4
+          a3: {^RandomInt: {min: 0, max: 100}}
+          b3: {^RandomInt: {min: 0, max: 100}}
+          c3: {^Array: {of: {^FastRandomString: {length: 11}}, number: 10}}
+        Indexes:
+        - keys: {b1: 1}
+        - keys: {c2: 1, d2: 1}
+
+# Ensure all data is synced to disk.
+- Name: Quiesce
+  Type: QuiesceActor
+  Threads: 1
+  Database: *Database
+  Phases:
+    OnlyActiveInPhases:
+      Active: [2]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Threads: 1
+
+# The simplifier optimizes out an $or's branch and makes an Index Scan plan possible instead of 
+# CollectionScan one
+- Name: SimplifiedOutOrBranch.BM
+  Type: CrudActor
+  Database: *Database
+  Threads: *Threads
+  Phases:
+    OnlyActiveInPhases:
+      Active: [3]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Duration: *Duration
+        Collection: Collection0
+        Operations:
+        - OperationName: find
+          OperationCommand:
+            Filter:  {$or: [{$and: [{a1: 1}, {a1: {$ne: 1}}]}, {b1: 2}]}
```

Comment:
Might be helpful to include the "simplified" query here for reference (so, just mention its {b1: 2}, in case its not obvious to someone). Could do the same for other queries.

Reply:
done